community/codegen/src/test/java/org/neo4j/codegen/CodeGenerationTest.java

```diff
@@ -1056,6 +1056,150 @@ public void shouldGenerateMethodUsingAnd() throws Throwable
         assertThat( conditional.invoke( false, false ), equalTo( false ) );
     }
 
+    @Test
+    public void shouldGenerateMethodUsingMultipleAnds() throws Throwable
+    {
+        // given
+        ClassHandle handle;
+        try ( ClassGenerator simple = generateClass( "SimpleClass" ) )
+        {
+            try ( CodeBlock conditional = simple.generateMethod( boolean.class, "conditional",
+                    param( boolean.class, "test1" ),
+                    param( boolean.class, "test2" ),
+                    param( boolean.class, "test3" ) ) )
+            {
+                conditional.returns( and(
+                        conditional.load( "test1" ),
+                        and( conditional.load( "test2" ),
+                                conditional.load( "test3" ) ) ) );
+            }
+
+            handle = simple.handle();
+        }
+
+        // when
+        MethodHandle conditional =
+                instanceMethod( handle.newInstance(), "conditional", boolean.class, boolean.class, boolean.class );
+
+        // then
+        assertThat( conditional.invoke( true, true, true ), equalTo( true ) );
+        assertThat( conditional.invoke( true, false, true ), equalTo( false ) );
+        assertThat( conditional.invoke( false, true, true ), equalTo( false ) );
+        assertThat( conditional.invoke( false, false, true ), equalTo( false ) );
+        assertThat( conditional.invoke( true, true, false ), equalTo( false ) );
+        assertThat( conditional.invoke( true, false, false ), equalTo( false ) );
+        assertThat( conditional.invoke( false, true, false ), equalTo( false ) );
+        assertThat( conditional.invoke( false, false, false ), equalTo( false ) );
```

Comment:
These 4 tests are great, if somewhat similar. Did you consider extracting common structure, or you just prefer it like this? I'm fine with either, just curious.

Reply:
I don't have a really good excuse, but we tend to allow more duplication in test code and this implementation is not expected to change very often.